stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -3505,3 +3505,13 @@ end
 (+)(A::SparseMatrixCSC, J::UniformScaling) = A + sparse(J, size(A)...)
 (-)(A::SparseMatrixCSC, J::UniformScaling) = A - sparse(J, size(A)...)
 (-)(J::UniformScaling, A::SparseMatrixCSC) = sparse(J, size(A)...) - A
+
+## circular shift
+
+function circshift!(O::SparseMatrixCSC, X::SparseMatrixCSC, (r,c)::Base.DimsInteger{2})
+    I, J, V = findnz(X)
+    O .= sparse(mod1.(I .+ r, X.n), mod1.(J .+ c, X.m), V, X.n, X.m)
```

Comment:
Why isn't this `X.m, X.n`?

Reply:
Sorry, I've been sloppy and didn't notice because I used square matrices. I modified the code and tests to catch this.

Reply:
(Make sure the non-square case has test coverage.)